### DISCUSSION TITLE
Spark Engine Support SQLSTATE to SQL Exception

### DISCUSSION THREAD

**Discussion by beryllw** [apache/kyuubi#6064] [Ideas]
# Motivation

SQLSTATE is SQL standard state, and spark support SQLSTATE from 3.2.0, we can support Spark Engine's SQLSTATE for SQL Exception in kyuubi.

found some previous works, but don’t know why not continued.
[Umbrella] Support SQLSTATE and venderCode in KyuubiSQLException #670
0A000: feature_not_supported #728

related spark work:
https://issues.apache.org/jira/browse/SPARK-34920
https://spark.apache.org/docs/latest/sql-error-conditions.html

sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServerErrors.scala
```scala
  def runningQueryError(e: Throwable, format: ErrorMessageFormat.Value): Throwable = e match {
    case st: SparkThrowable if format == ErrorMessageFormat.PRETTY =>
      val errorClassPrefix = Option(st.getErrorClass).map(e => s"[$e] ").getOrElse("")
      new HiveSQLException(
        s"Error running query: $errorClassPrefix${st.toString}", st.getSqlState, st)
    case st: SparkThrowable with Throwable =>
      new HiveSQLException(SparkThrowableHelper.getMessage(st, format), st.getSqlState, st)
    case _ => new HiveSQLException(s"Error running query: ${e.toString}", e)
  }
```

**Comment by beryllw** [comment]
@yaooqinn @pan3793 cc

**Reply by pan3793** [reply]
let's reboot this task!
